packages/app/src/cli/services/app-logs/dev/write-app-logs.test.ts

```diff
@@ -27,40 +41,63 @@ describe('writeAppLogsToFile', () => {
     stdout = {write: vi.fn()}
   })
 
-  test('calls writeLog with the right data', async () => {
+  test('calls writeLog with the FunctionRunLog payload type', async () => {
     // Given
     // determine the fileName and path
     const fileName = `20240522_150641_827Z_${APP_LOG.source_namespace}_${APP_LOG.source}`
     const path = joinPath(API_KEY, fileName)
 
     // When
-    const returnedPath = await writeAppLogsToFile({appLog: APP_LOG, apiKey: API_KEY, stdout})
+    const returnedPath = await writeAppLogsToFile({
+      appLog: APP_LOG,
+      appLogPayload: FUNCTION_RUN_PAYLOAD,
+      apiKey: API_KEY,
+      stdout,
+    })
 
     // Then
     expect(returnedPath.fullOutputPath.startsWith(path)).toBe(true)
-    expect(writeLog).toHaveBeenCalledWith(expect.stringContaining(path), expectedLogDataFromAppEvent(APP_LOG))
+    expect(writeLog).toHaveBeenCalledWith(
+      expect.stringContaining(path),
+      expectedLogDataFromAppEvent(APP_LOG, FUNCTION_RUN_PAYLOAD),
+    )
   })
 
-  test('prints and re-throws parsing errors', async () => {
+  test('calls writeLog with strings when no matching payload type', async () => {
     // Given
-    const appLog = {
-      ...APP_LOG,
-      payload: 'invalid JSON',
-    }
+    // determine the fileName and path
+    const fileName = `20240522_150641_827Z_${APP_LOG.source_namespace}_${APP_LOG.source}`
+    const path = joinPath(API_KEY, fileName)
+
+    // When
+    const returnedPath = await writeAppLogsToFile({
+      appLog: NEW_APP_LOG,
+      appLogPayload: JSON.parse(NEW_APP_LOG.payload),
+      apiKey: API_KEY,
+      stdout,
+    })
 
-    // When/Then
-    await expect(writeAppLogsToFile({appLog, apiKey: API_KEY, stdout})).rejects.toThrow()
-    expect(stdout.write).toHaveBeenCalledWith(expect.stringContaining('Error while writing log to file: '))
+    // Then
+    expect(returnedPath.fullOutputPath.startsWith(path)).toBe(true)
+    expect(writeLog).toHaveBeenCalledWith(
+      expect.stringContaining(path),
+      expectedLogDataFromAppEvent(NEW_APP_LOG, JSON.parse(NEW_APP_LOG.payload)),
+    )
   })
 })
 
-function expectedLogDataFromAppEvent(event: AppLogData): string {
+function expectedLogDataFromAppEvent(event: AppLogData, payload: AppLogPayload | string): string {
   const data = camelcaseKeys(
     {
       ...event,
-      payload: JSON.parse(event.payload),
+      payload: payload as any,
     },
     {deep: true},
   )
+
+  if (payload instanceof FunctionRunLog) {
+    data.payload.logs = payload.logs.split('\n').filter(Boolean)
+  }
```

Comment:
This doesn't test much given that it's just the same code as the module it's testing- should we test the full output for these tests instead now that the conversion from input to output is non-trivial?